metabase/metabase
# Use metabase-lib v2 to manage order-bys in the notebook editor

Resolves #29842, epic #28034

Makes the notebook editor use metabase-lib v2 (MLv2) to manage order-bys (aka sorts). Most changes are concentrated in the `SortStep` component and `metabase-lib/order_by` files, but this PR also introduces some shared code to deal with columns in MLv2 (e.g. changes to `metabase-lib/metadata`, shared `QueryColumnPicker` component)

### To Verify

1. Open `/question/:id/notebook`
2. Ensure sorting works (try sorting, changing sorting direction, try using columns from source/joined table/question/model, etc.)


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/29842)
